TestIncrCounter disabled integration/walrus test

```
$ SG_TEST_BACKING_STORE=Couchbase SG_TEST_COUCHBASE_SERVER_URL=http://localhost:8091 SG_TEST_USE_XATTRS=false ./test.sh -run ^TestIncrCounter$
Testing code with 'go test' ...
Integration mode: forcing tests to run in serial across packages via -p 1 flag
Running Sync Gateway unit tests
?   	github.com/couchbase/sync_gateway	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/couchbase/sync_gateway/auth	0.016s
=== RUN   TestIncrCounter
2017-11-06T16:57:48.796-08:00 NOT Using xattrs: strings.ToLower(useXattrs) != strings.ToLower(TestEnvSyncGatewayTrue).  |false| != |true|
2017-11-06T16:57:53.282-08:00 GoCBCustomSGTranscoder Opening Couchbase database test_data_bucket on <http://localhost:8091> as user "test_data_bucket"
--- FAIL: TestIncrCounter (4.50s)
	bucket_gocb_test.go:472: Attempt to retrieve non-existent counter should return error
FAIL
exit status 1
FAIL	github.com/couchbase/sync_gateway/base	4.517s
```

Original issue: https://github.com/couchbase/sync_gateway/issues/2977

--- a/base/bucket_gocb.go
+++ b/base/bucket_gocb.go
@@ -1567,6 +1567,10 @@ func (bucket CouchbaseBucketGoCB) WriteWithXattr(k string, xattrKey string, exp 
 	}
 }
 
+// Increment the atomic counter k by amt.
+//
+// - If amt is 0 and the atomic counter for that key exists, this is treated as a GET operation that returns the current value.
+// - If amt is 0 but the key does not exist, then it will return 0
 func (bucket CouchbaseBucketGoCB) Incr(k string, amt, def uint64, exp uint32) (uint64, error) {
 
 	// GoCB's Counter returns an error if amt=0 and the counter exists.  If amt=0, instead first

--- a/base/bucket_gocb_test.go
+++ b/base/bucket_gocb_test.go
@@ -430,8 +430,6 @@ func TestUpdate(t *testing.T) {
 
 func TestIncrCounter(t *testing.T) {
 
-	t.Skip("Currently not passing: under walrus, go-couchbase and gocb, this test results in: Attempt to retrieve non-existent counter should return error")
-
 	testBucket := GetTestBucketOrPanic()
 	defer testBucket.Close()
 	bucket := testBucket.Bucket
@@ -466,11 +464,7 @@ func TestIncrCounter(t *testing.T) {
 	}
 	assert.Equals(t, retrieval, uint64(2))
 
-	// Attempt retrieval of a non-existent counter using delta=0
-	retrieval, err = bucket.Incr("badkey", 0, 0, 0)
-	if err == nil {
-		t.Errorf("Attempt to retrieve non-existent counter should return error")
-	}
+
 
 }
 

